src/BlockchainWriter/Controller.ts

```diff
@@ -87,7 +87,7 @@ export class Controller {
     const data = poetAnchorToData(poetAnchor)
     const txId = await anchorData(data)
 
-    await dao.setTransactionId(ipfsDirectoryHash, txId)
+    await dao.setTransactionId(ipfsDirectoryHash, txId, new Date().toISOString())
```

Comment:
Why pass the third argument when the function provides the same value as a default?

Reply:
It might be argued that the controller is in control of the data and should be responsible for passing in a date even if the dao provided a default. IDC either way, just thought it was good to be explicit in this case.

Reply:
Then the default should be removed so it's explicit in every case right?

Reply:
I wouldn't say `should`. If this were JavaScript I would have used a valid default to provide a type hint. In TS I could just make it a required type of string, but I still prefer valid defaults.

Reply:
In TS I guess requiring a type of string would act more like argument validation in a limited way, but I am not really a fan of that, I would rather add real validation checking.

Reply:
But I don't mind changing this up if you and the other prefer a change to this.

Reply:
actually, I see why this is a bit awakard after thinking about it a bit more, the dao function should be a bit more general and should know the least amount posssible as about the controller requirements, so I should probably update and rename the function to take an updated entry. `findAndUpdateByIPFSDirectoryHash` would make more sense, that allows the controller just to pass updates in as an object.

Reply:
@warrenv I've updated it, let me know what you think.

Reply:
Yes, that makes more sense now.